components/eam/src/physics/crm/physpkg.F90

```diff
@@ -1,1802 +1,1177 @@
+  !  aerosol dry deposition processes
+  !-----------------------------------------------------------------------------
+  if (l_tracer_aero) then
 
-    !  aerosol dry deposition processes
-    call t_startf('aero_drydep')
     call aero_model_drydep( state, pbuf, obklen, surfric, cam_in, ztodt, cam_out, ptend )
     call physics_update(state, ptend, ztodt, tend)
-    call t_stopf('aero_drydep')
 
-    !---------------------------------------------------------------------------------
-    !  ... enforce charge neutrality
-    !---------------------------------------------------------------------------------
+    ! enforce charge neutrality
     call charge_fix( ncol, state%q(:,:,:) )
 
-end if ! l_tracer_aero
+  end if ! l_tracer_aero
 
-if (l_gw_drag) then
-    !===================================================
-    ! Gravity wave drag
-    !===================================================
-    call t_startf('gw_tend')
+  !-----------------------------------------------------------------------------
+  ! Gravity wave drag
+  !-----------------------------------------------------------------------------
+  if (l_gw_drag) then
 
     call gw_tend(state, sgh, pbuf, ztodt, ptend, cam_in)
-
     call physics_update(state, ptend, ztodt, tend)
+    
     ! Check energy integrals
     call check_energy_chng(state, tend, "gwdrag", nstep, ztodt, zero, zero, zero, zero)
-    call t_stopf('gw_tend')
-
-    ! QBO relaxation
-    call qbo_relax(state, pbuf, ptend)
-    call physics_update(state, ptend, ztodt, tend)
-    ! Check energy integrals
-    call check_energy_chng(state, tend, "qborelax", nstep, ztodt, zero, zero, zero, zero)
 
-    ! Ion drag calculation
-    call t_startf ( 'iondrag' )
+  end if ! l_gw_drag
 
-    if ( do_waccm_ions ) then
-       call iondrag_calc( lchnk, ncol, state, ptend, pbuf,  ztodt )
-    else
-       call iondrag_calc( lchnk, ncol, state, ptend)
-    endif
-    !----------------------------------------------------------------------------
-    ! Call ionosphere routines for extended model if mode is set to ionosphere
-    !----------------------------------------------------------------------------
-    if( waccmx_is('ionosphere') ) then
-       call ionos_intr(state, ptend, pbuf, ztodt)
-    endif
-
-    call physics_update(state, ptend, ztodt, tend)
-    ! Check energy integrals
-    call check_energy_chng(state, tend, "iondrag", nstep, ztodt, zero, zero, zero, zero)
-    call t_stopf  ( 'iondrag' )
-
-end if ! l_gw_drag
-
-
-if (l_ac_energy_chk) then
-    !-------------- Energy budget checks vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv
+  !-----------------------------------------------------------------------------
+  ! Energy budget checks
+  !-----------------------------------------------------------------------------
+  if (l_ac_energy_chk) then
 
     call pbuf_set_field(pbuf, teout_idx, state%te_cur, (/1,itim_old/),(/pcols,1/))       
 
     tmp_t(:ncol,:pver) = state%t(:ncol,:pver)
```

Comment:
I'm not sure what this is used for, but this looks like a potential problem both in this PR and in the original version; `tmp_t` is only set within the logical `if (l_ac_energy_chk)`, but is used regardless, *outside* this logical, in the below call to `diag_phys_tend_writeout`. Likewise for the other `tmp_` variables...

Reply:
so I looked into this more and it's kinda weird. It seems that all these temp variables mostly don't matter when using the spectral element. The tmp_t variable is used to calculate a TTEND residual, but as far as I can tell it will always be zero unles set_dry_to_wet() or physics_dme_adjust() has a significant effect on the temperature. 

Anyway, I dont want to tackle the weirdness of those diagnostics, but to address the concern you raised we can simply move diag_phys_tend_writeout() inside the if block for l_ac_energy_chk. That will avoid sending uninitialized data into diag_phys_tend_writeout.

Reply:
Actually... looking back at this I think it's better to remove the check for l_ac_energy_chk entirely. The flag suggests it's just "checking" the energy change, but the block includes set_dry_to_wet() which actually changes the constituent mixing ratios to be consistent with the dycor. Similarly there's a dry mass adjustment, physics_dme_adjust(), that is also modifying the state.

Reply:
ok, I removed the check for l_ac_energy_chk since this is the default behavior and the concerns in my previous comment.